src/Http/Http/test/ResponseCookiesTest.cs

```diff
@@ -74,6 +75,37 @@ public void DeleteCookieShouldSetDefaultPath()
             Assert.Contains("expires=Thu, 01 Jan 1970 00:00:00 GMT", cookieHeaderValues[0]);
         }
 
+        [Fact]
+        public void DeleteCookieWithDomainAndPathShouldSetDefaultPath()
+        {
+            var headers = (IHeaderDictionary)new HeaderDictionary();
+            var features = MakeFeatures(headers);
+            var responseCookies = new ResponseCookies(features);
+
+            var testCookies = new (string Key, string Path, string Domain)[]
+            {
+                new ("key1", "/", null),
+                new ("key1", "/test/", null),
+                new ("key2", "/", "localhost"),
+                new ("key2", "/test/", "localhost"),
+            };
+
+            foreach (var cookie in testCookies)
+            {
+                responseCookies.Delete(cookie.Key, new CookieOptions() { Domain = cookie.Domain, Path = cookie.Path });
+            }
+
+            var cookieHeaderValues = headers.SetCookie.ToArray();
+            Assert.Equal(testCookies.Length, cookieHeaderValues.Length);
+
+            var deletedCookies = cookieHeaderValues.ToArray();
+            Assert.Contains(deletedCookies, cookie => cookie.StartsWith("key1", StringComparison.InvariantCulture) && cookie.Contains("path=/"));
```

Comment:
Can't test for just "path=/" because it matches both cookies, so we're not actually testing that both are there 😄

Reply:
Effectively 🙂.
In this case, we can improve the testing with specific path  **`"/path1/", "path2/"`**,and test the matching.